pkg/volume/csi/csi_plugin.go

```diff
@@ -888,3 +990,15 @@ func isV1Version(version string) bool {
 
 	return parsedVersion.Major() == 1
 }
+
+// ToPluginHandler converts the generic volume plugin to a specific csiPlugin and
+// returns that as a PluginHandler. In case the generic plugin is not a CSI
+// plugin and thus not a PluginHandler, an error is returned.
+func ToPluginHandler(genericPlugin volume.VolumePlugin) (pluginwatcher.PluginHandler, error) {
```

Comment:
I don't think this is needed since the code asserts that type `csiPlugin` is in fact a PluginHandler.  If there's a location where the interface `volume.Plugin` is used, please use the concrete type `csiPlugin` to avoid this conversion.  If there is a need for it, please point it out.

Reply:
Just to be clear, this function is not used yet, we just bring it in place, so we can use it for the later [PR] / [commit], which deals with the kubelet changes. The idea here is to get as much in place up front as we can, so the change to the kubelet is it own, rel. small PR.

[PR]: https://github.com/kubernetes/kubernetes/pull/74963
[commit]: https://github.com/kubernetes/kubernetes/pull/74963/commits/8401876b75920745629c7f062af518bc7a1e4927

The purpose of this series of PRs is to get rid of sharing state with external things (e.g. the kubelet) via exporting a package global variable, the `csi.PluginHandler`.

However, the kubelet still needs to have access to the `PluginHandler` aspects of the cis plugin. We can do this by finding the csi plugin **by name** from the list of volume plugins the kubelet has already registered. But doing that, gives us the plugin as a generic `volume.VolumePlugin`. So we came up with two options:
1. do and check the type assertion in the kubelet
1. have this little utility function available which manages the type assertion

We went for the second option here, because the first option would mean that we need to export `csi.csiPlugin` and push more of the "responsibility" into the kubelet, when it is actually a feature of the `csiPlugin`, to be used as a `PluginHandler`.

Hopefully that makes it clearer what we are trying to do here?